tests/core/circuit/test_circuit.py

```diff
@@ -57,6 +59,19 @@ def test_three_qubit_gate():
     assert len(qc.gates) == 1
     assert qc.gates[0].gate == ThreeQubitGateType.CCX
 
+def test_exp_evolution():
+    hamiltonian = qm_o.Hamiltonian()
+    hamiltonian += qm_o.X(0) * qm_o.Z(1)
+    qc = QuantumCircuit(2)
+    theta = Parameter("theta")
+    qc.exp_evolution(theta,hamiltonian)
+    assert len(qc.gates) == 1
+    assert isinstance(qc.gates[0], ParametricExpGate)
+    assert qc.gates[0].parameter == theta
+    assert len(qc.gates[0].indices) == 2
+    assert qc.gates[0].hamiltonian == hamiltonian
+
+
```

Comment:
It is better to add raise error test.

Reply:
Do you mean something like ` assert qc.gates[0].hamiltonian == hamiltonian , "The Hamiltonian in the gate does not match with the Hamiltoina defined"`? Also, should I add to all `assert` statement ?

Reply:
Not sure what `raise error test` is like?